src/rules/node.ts

```diff
@@ -48,40 +69,60 @@ export let node = {
 
   /** Warns when 'console.log' is added to code */
   async consoleLog() {
-    const files = ([] as string[]).concat(
-      danger.git.created_files  || [],
-      danger.git.modified_files || [],
-    );
-
-    // tslint:disable-next-line:prefer-for-of
-    for (let i = 0; i < files.length; i++) {
-      const file = await danger.git.diffForFile(files[i]);
-      if (file && file.added.match(/console\.log/ig)) {
-        warn('This PR adds `console.log` to code.');
-        // breaks to avoid reading all files when we already know that there is an issue
-        break;
-      }
+    if (changedFilesContainsRegex('console\.log')) {
+      warn('This PR adds `console.log` to code.');
     }
   },
 
   /** Warns when 'npm install -g' is added to code */
   async npmGlobal() {
+    if (await changedFilesContainsRegex('npm install \\-g')) {
+      // tslint:disable-next-line:max-line-length
+      fail(['This PR adds `npm install \-g` instuctions to the project, ',
+        'please prefer adding packages as `dependencies` or `devDependencies` to `package.json` instead. ',
+        'More info at https://github.com/typeorm/typeorm/pull/599'].join(''));
+    }
+  },
+
+  /** Warns when node version is changed */
+  async nodeVersion() {
+
     const files = ([] as string[]).concat(
       danger.git.created_files  || [],
       danger.git.modified_files || [],
     );
 
-    // tslint:disable-next-line:prefer-for-of
-    for (let i = 0; i < files.length; i++) {
-      const file = await danger.git.diffForFile(files[i]);
-      if (file && file.added.match(/npm install \-g/ig)) {
-        // tslint:disable-next-line:max-line-length
-        fail(['This PR adds `npm install \-g` instuctions to the project, ',
-          'please prefer adding packages as `dependencies` or `devDependencies` to `package.json` instead. ',
-          'More info at https://github.com/typeorm/typeorm/pull/599'].join(''));
-        // breaks to avoid reading all files when we already know that there is an issue
-        break;
+    const packageJsonNodeVersion = await (async () => {
+      const packageJsonIndex = files.findIndex(fileName => fileName.match(/package\.json/gi) != null);
+      let packageJsonNodeVersionValue;
+      if (packageJsonIndex > -1) {
+        const packageJsonFile = await danger.git.JSONDiffForFile(files[packageJsonIndex]);
+        if (packageJsonFile['engines.node']) {
+          packageJsonNodeVersionValue = packageJsonFile['engines.node'].after;
+        }
+      }
+      return packageJsonNodeVersionValue;
+    })();
+
+    const nvmrcNodeVersion = await (async () => {
+      let nvmrcNodeVersionValue;
+      const nvmrcIndex = files.findIndex(fileName => fileName.match(/\.nvmrc/gi) != null);
+      if (nvmrcIndex > -1) {
+        const nvmrcFile = await danger.git.diffForFile(files[nvmrcIndex]);
+        if (nvmrcFile) {
+          nvmrcNodeVersionValue = nvmrcFile.after;
+        }
       }
+      return nvmrcNodeVersionValue;
+    })();
```

Comment:
I guess this could be changed to

```
const nvmrcFile = await files
  .filter(name => name.match(/\.nvmrc/gi)) // ['file 1', 'file 2', 'file 3'] or []
  .filter((_, index) => index === 0) // ['file 1'] or []
  .map(name => danger.git.diffForFile(name)) // [Promise] or []
  .find((_, index) => index === 0) || Promise.resolve(null); // Promise

if (nvmrcFile) {
  return nvmrcFile.after;
} else {
  return null;
}

```